src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -30,52 +42,105 @@ def execute(self, reqStatus):
             self.logger.debug('+++ monit found %s requests in %s state',
                               len(requests), reqStatus)
 
-            # FIXME: completion threshold should come either from unified
-            # or campaign configuration, e.g.
-            # thr = self.unifiedConfig().get('completedThreshold', 100)
-            thr = 100
-
-            requestStatuses = {}  # keep track of request statuses
+            # get list of all transfer records
+            transferRecords = [d for d in self.getTransferInfo('ALL_DOCS')]
+            # keep track of completed workflows
+            completedWorkflows = []
+            # keep track of request and their new statuses
+            reqStatus = {}
+            # main logic
             for reqName in requests:
-                req = {'name': reqName, 'reqStatus': reqStatus}
-                self.logger.debug("+++ request %s", req)
-                # obtain status records from couchdb for given request
-                transferRecords = self.getTransferRecords(reqName)
-                if not transferRecords:
+                self.logger.debug("+++ request %s", reqName)
+                # for us workflow name is the same as request name
+                wname = reqName
+                # obtain transfer records for our request
+                transfers = []
+                for rec in transferRecords:
+                    if reqName in rec:
```

Comment:
Isn't `rec` the transfer record? If so, it doesn't have a key named after the workflow.

Reply:
done